app/controllers/submissions_controller.rb

```diff
@@ -431,8 +432,10 @@ def get_file
           file_contents = repo.download_as_string(raw_file)
           file_contents.encode!('UTF-8', invalid: :replace, undef: :replace, replace: '�')
 
-          if file_type == 'markusurl'
+          if file_type == 'markusurl' && assignment.url_submit
```

Comment:
This can be simplified:

```ruby
if file_type == 'markusurl'
  file_contents = assignment.url_submit ? extract_url(file_contents) : 'unknown'
end
```